apps/api/src/employee/employee.service.ts

```diff
@@ -43,24 +51,83 @@ export class EmployeeService {
     return this.ienUsersRepository.findOne({ email });
   }
 
+  _nameSearchQuery(keyword: string) {
+    let keywords = keyword.split(' ');
+    keywords = keywords.filter(item => item.length);
+    if (keywords.length === 1) {
+      return `(Employee.name ilike '%${keywords[0].toLowerCase()}%')`;
+    } else if (keywords.length === 2) {
+      return `(Employee.name ilike '%${keywords[0]}%${keywords[1]}%' OR Employee.name ilike '%${keywords[1]}%${keywords[0]}%')`;
+    } else if (keywords.length === 3) {
+      const possibleShuffle = [];
+      possibleShuffle.push(`Employee.name ilike '%${keywords[0]}%${keywords[1]}%${keywords[2]}%'`);
+      possibleShuffle.push(`Employee.name ilike '%${keywords[0]}%${keywords[2]}%${keywords[1]}%'`);
+      possibleShuffle.push(`Employee.name ilike '%${keywords[1]}%${keywords[0]}%${keywords[2]}%'`);
+      possibleShuffle.push(`Employee.name ilike '%${keywords[1]}%${keywords[2]}%${keywords[0]}%'`);
+      possibleShuffle.push(`Employee.name ilike '%${keywords[2]}%${keywords[0]}%${keywords[1]}%'`);
+      possibleShuffle.push(`Employee.name ilike '%${keywords[2]}%${keywords[1]}%${keywords[0]}%'`);
+      return `( ${possibleShuffle.join(' OR ')} )`;
+    }
+    return `Employee.name ilike '%${keyword}%'`;
+  }
+
   /**
    * List and filter employees,
    * Only for administrator purposes
    * We will build UI for this when we extend our scope to provide user-management
    * @param name optional name wise filter
    * @returns Employee/User's list
    */
-  async getEmployeeList(name: string): Promise<EmployeeEntity[]> {
-    if (!name) {
-      name = '';
+  async getEmployeeList(filter: EmployeeFilterAPIDTO) {
+    const { role, name, sortKey, order, limit, skip } = filter;
+    const query: FindManyOptions<EmployeeEntity> = {
+      order: {
+        [sortKey || 'createdDate']: sortKey ? order : 'DESC',
+      },
+    };
+
+    if (limit) query.take = limit;
+    if (skip) query.skip = skip;
+
+    if (!role && !name) {
+      return this.employeeRepository.findAndCount(query);
     }
-    return getManager()
-      .createQueryBuilder(EmployeeEntity, 'employee')
-      .select('employee.id, employee.name, employee.role, employee.email, employee.created_date')
-      .addSelect('users.id', 'user_id')
-      .leftJoin('ien_users', 'users', 'employee.email = users.email')
-      .where('employee.name ilike :name', { name: `%${name}%` })
-      .getRawMany();
+
+    const conditions: (string | ObjectLiteral)[] = [];
+
+    if (role) {
```

Comment:
on purpose?

Reply:
yup, once role and name filter is implemented, that's what these are for